### PR TITLE
Only report test failures if they didn't succeed after a retry

### LINKED ISSUE #105: Verify re-run tests that succeed after failure are not blocking PRs
A test can be re-run after failure and succeeding:
image

Though, it looks like PRs still get blocked by a test failure being reported. We should look into this.

It's best to reproduce this locally by writing a test that fails 50/50

Example log:
[apps_6bee37a5a2a41cdb_47ac49f1-faed-438b-a7e7-b1bb900d7b0a-full.txt.zip](https://github.com/WeTransfer/WeTransfer-iOS-CI/files/7546321/apps_6bee37a5a2a41cdb_47ac49f1-faed-438b-a7e7-b1bb900d7b0a-full.txt.zip)

### PR DESCRIPTION
**Pull request by AvdLee**
As a start, this PR makes sure we don't report test failures if they eventually succeed after retrying.
I've also added a project which we can use to create sample `xcresult` files for testing.

I'm going to add extra logging in a future PR in regards to skipped and retried tests.

Fixes #105 

_Note: There are a lot of files, but most of them are just the `xcresult` file chunks_

### CODE REVIEW

File: WeTransferPRLinter/Package.swift [hunk 1]
```diff
@@ -17,8 +17,8 @@ let package = Package(
     dependencies: [
         // Local package is required for development since Danger does not support Swift 5.5 correctly yet with Danger-Plugin testing.
 //         .package(name: "danger-swift", path: "../../../../../Forks/swift"),
-        .package(name: "danger-swift", url: "https://github.com/danger/swift.git", from: "3.12.1"),
-//        .package(name: "danger-swift", url: "https://github.com/AvdLee/swift.git", .branch("master")),
+//        .package(name: "danger-swift", url: "https://github.com/danger/swift.git", from: "3.12.1"),
+        .package(name: "danger-swift", url: "https://github.com/AvdLee/swift.git", .branch("master")),
```
[the remark this answers is not in the document]
Comment (kevinrenskers) [reply WeTransferPRLinter/Package.swift:21]: Would it maybe be better to have that fork on the WeTransfer org vs your personal account? It adds to the bus factor, so to speak.

**Review by kevinrenskers — APPROVED** [review]

File: WeTransferPRLinter/Package.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (AvdLee) [reply WeTransferPRLinter/Package.swift:21]: Kaira asked that before too. However, I've had this fork ready and it should be a temporary solution either way. The linked PR can be merged anytime at which we move back to the main repo

[the remark this answers is not in the document]
Comment (kevinrenskers) [reply WeTransferPRLinter/Package.swift:21]: Still, forking your fork to the WeTransfer org wouldn't take any effort either, but fair enough.

[the remark this answers is not in the document]
Comment (AvdLee) [reply WeTransferPRLinter/Package.swift:21]: Hmm, good point! I wasn't aware that's something you can do, but that might actually be easy. I'll do that next time we need this

**Pull request merged**